bccsp/pkcs11/pkcs11_test.go

```diff
@@ -283,6 +296,43 @@ func TestECDSAGetKeyBySKI(t *testing.T) {
 	require.Equalf(t, k.SKI(), k2.SKI(), "expected %x got %x", k.SKI(), k2.SKI())
 }
 
+func TestAlternateLabelGeneration(t *testing.T) {
+	// We generate a unique Alt ID for the test
+	uniqueAltId := strconv.FormatInt(time.Now().UnixNano()/1e6, 10)
+	fakeSKI := []byte("FakeSKI")
+
+	opts := defaultOptions()
+	opts.AltID = uniqueAltId
+	opts.Immutable = true
+
+	csp, cleanup := newProvider(t, opts)
+	defer cleanup()
+
+	sess, err := csp.getSession()
+	require.NoError(t, err)
+	defer csp.returnSession(sess)
+
+	// Passing fake SKI to ensure that the look up fails if the uniqueAltId is not used
+	k, err := csp.findKeyPairFromSKI(sess, fakeSKI, privateKeyType)
+	if err == nil {
+		t.Fatalf("Found a key when expected to find none.")
+	}
+
+	// Now generate a new EC Key, which should be using the uniqueAltId
+	_, _, err = csp.generateECKey(oidNamedCurveP256, true)
+	if err != nil {
+		t.Fatalf("Failed generating Key [%s]", err)
+	}
+
+	k, err = csp.findKeyPairFromSKI(sess, fakeSKI, privateKeyType)
+	if err != nil {
+		t.Fatalf("Found no key after generating an EC Key based on the AltId.")
+	}
+	if k == 0 {
```

Comment:
This is saying we got _some_ object handle back. It's not asserting on the attributes.